### ISSUE TITLE
Add support for inline templateSpec templates

### ISSUE THREAD

**Issue by bmoore-msft** [microsoft/vscode-azurearmtools#1128]
templateSpecs allow for an inline template similar to nested templates - and would require an inner scope eval.  Currently we treat these templates as part of the parent the way we used to handle nested templates.  Sample:

```
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "resources": [
    {
      "type": "Microsoft.Resources/templateSpecs",
      "apiVersion": "2019-06-01-preview",
      "name": "test",
      "location": "[resourceGroup().location]",
      "properties": {
        "description": "Sample",
        "displayName": "Sample"
      },
      "resources": [
        {
          "type": "versions",
          "apiVersion": "2019-06-01-preview",
          "name": "1.0.0",
          "properties": {
            "description": "Sample",
            "template": {
              "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
              "contentVersion": "1.0.0.0",
              "parameters": {
                  "name": {
                      "type": "string"
                  }
              },
              "resources": [
                {
                  "apiVersion": "2018-02-01",
                  "name": "[parameters('name')]",
                  "type": "Microsoft.Resources/deployments",
                  "properties": {
                    "mode": "Incremental",
                    "template": {
                      "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
                      "contentVersion": "1.0.0.0",
                      "resources": []
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  ]
}
```

Note the param validation

**Comment by StephenWeatherford** [comment]
Is this currently supported?  Where do I find doc on how to know when to use inner scope eval?

**Comment by bmoore-msft** [comment]
Docs here: https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-specs?tabs=azure-powershell

But essentially any time you see this property:
https://github.com/Azure/azure-quickstart-templates/blob/master/101-templatespec-create/prereqs/prereq.azuredeploy.json#L47

In this resource:
https://github.com/Azure/azure-quickstart-templates/blob/master/101-templatespec-create/prereqs/prereq.azuredeploy.json#L40

We'd trigger the same behavior as inner scope eval.

**Comment by alex-frankel** [comment]
@bmoore-msft, we don't have a scopeEval option for templateSpecs today. Would the proposal to be to use that logic whenever we see double square brackets (`[[...]`)?

**Comment by bmoore-msft** [comment]
No, trigger the logic whenever we are on a Microsoft.Resources/templateSpecs/version resource... trigger it for the template property just like we do for the deployments resource (when scope eval is inner on the deployments resource).

IOW, same behavior, just trigger on a new resourceType.

**Comment by alex-frankel** [comment]
I am still not following. Today a templateSpec in a template behaves the same as an "outer"-scoped nested template effectively. There is no way to achieve the equivalent of "inner" behavior, even if the tooling were able to treat it that way. In other words, using the `[[` trick is the only option you have.

**Comment by bmoore-msft** [comment]
The tooling doesn't help with the template part of the templateSpec - so the normal things you'd get with that (like parameter validation) you don't get with a templateSpec.  So in the case of an unescaped expression (e.g. param) there isn't any thing that tells you want the right fix might be.  In the example above, it says the param isn't defined, but it is, we're just looking in the wrong place.  Now the fix may very well be different as you noted, since that param, as used, needs to be escaped...

So perhaps not as simple as in line but the same idea.  Help me validate the templateSpec template as well, like we help validate the inner scoped template.

It may be worth a discussion about whether we should have inner scope for templateSpec versions, so that all the expressions need not be escaped... that alone makes templateSpecs all but unusable in a template.
